### ISSUE TITLE
How to deal with the negative loss?

### ISSUE THREAD

**Issue by Mrils** [mattpoggi/mono-uncertainty#1]
Hi!
     When I try to increase the uncertainty part (described as the **log** section in your paper) in the network, the loss changed to negative.
     Is this normal？ How did you solve it?

**Comment by minygd** [comment]
I think it's fine, as `log(x)` can always have a negative value when `x < 1`.

**Comment by Mrils** [comment]
So what is the direction of network optimization? 
Minimize the absolute value of network loss?

**Comment by mattpoggi** [comment]
Hi @Mrils 
that's correct. The optimization process pushes the denominator in Eq. 12,13 to be high, thus avoiding both the second term and the total loss to reach -infinite.

**Comment by Mrils** [comment]
So will the loss stabilize around a negative value after some epochs?

**Comment by mattpoggi** [comment]
Correct. In our experiments, this occurred quite early

**Comment by Mrils** [comment]
Got it! Thank you

**Comment by jilner** [comment]
Dear Mrils, I have  trouble writing the loss function when reproduction the training, could you share me the loss function codes  and I would be very grateful to you!
